src/exe/FreeC/Application/Options/Descriptors.hs

```diff
@@ -75,4 +75,20 @@ optionDescriptors =
     ++ "This option adds location information to error messages that\n"
     ++ "refer to the dumped file."
     )
+  , Option
+    []
+    ["from"]
+    (ReqArg (\p opts -> opts { optFrontend = p }) "LANG")
+    ("Optional. Specifies the frontend for the compiler to use. Currently the\n"
+    ++ "Haskell frontend and a dummy IR frontend that parses the IR are\n"
+    ++ "supported. Defaults to the Haskell frontend."
```

Comment:
The help message currently does not tell the user which values are actually allowed for the `--from` and `--to` command line options. For example, they might be surprised to find that `--from IR` is invalid even though IR is listed as supported. Furthermore, the help text has to be maintained manually at the moment. Thus, I'd suggest to generate the help text from the `name` fields of `backends` and `frontends`. The automatically generated help text could look as follows, for example.

```
Optional. Specifies the frontend for the compiler to use.
Allowed values are: `haskell`, `ir`.
Defaults to `haskell`.
```